src/newTest/java/org/opensearch/test/framework/TestIndex.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021-2022 floragunn GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test.framework;
+
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.exists.indices.IndicesExistsRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+
+public class TestIndex {
+
+    private final String name;
+    private final Settings settings;
+    
+    public TestIndex(String name, Settings settings) {
+        this.name = name;
+        this.settings = settings;
+        
+    }
+
+    public void create(Client client) {
+        if (!client.admin().indices().exists(new IndicesExistsRequest(name)).actionGet().isExists()) {
```

Comment:
Shouldn't create only be called when an index should be created, if we ignore already existing indexes we could be exposing a bug, what do you think?

If we keep this, rather than make 2 service requests if the index doesn't exist, could we instead create a single request and check the result and if `index already exists` error response comes back ignore it?

Reply:
Absolutely.

Reply:
I removed the check, so if the index already exists we should see an error. In my opinion there is no real use case for trying to create an index twice. If this happens, it is most likely an error.